httpclient-jetty/src/main/java/io/fabric8/kubernetes/client/jetty/JettyAsyncResponseListener.java

```diff
@@ -72,7 +72,11 @@ public void onHeaders(Response response) {
 
   @Override
   public void onComplete(Result result) {
-    asyncBodyDone.complete(null);
+    if (result.isSucceeded()) {
+      asyncBodyDone.complete(null);
+    } else {
+      asyncBodyDone.completeExceptionally(result.getRequestFailure());
```

Comment:
Is the intention to also have onFailure overriden? Because there could still be a response failure. From the javadocs, the advantage of onFailure is that it can be called sooner in some circumstances, like large file uploads.